crates/oxc_linter/src/rules/typescript/consistent_type_definitions.rs

```diff
@@ -0,0 +1,214 @@
+    ///
+    ///
+    /// ### Why is this bad?
+    ///
+    ///
+    /// ### Example
+    /// ```javascript
+    /// ```
+    ConsistentTypeDefinitions,
+    style
+);
+
+impl Rule for ConsistentTypeDefinitions {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let config = value.get(0).and_then(serde_json::Value::as_str).map_or_else(
+            ConsistentTypeDefinitionsConfig::default,
+            |value| match value {
+                "type" => ConsistentTypeDefinitionsConfig::Type,
+                _ => ConsistentTypeDefinitionsConfig::Interface,
+            },
+        );
+        Self { config }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::TSTypeAliasDeclaration(typ) => match &typ.type_annotation {
+                TSType::TSTypeLiteral(lit)
+                    if self.config == ConsistentTypeDefinitionsConfig::Interface =>
+                {
+                    ctx.diagnostic(ConsistentTypeDefinitionsDiagnostic("interface", "type", lit.span))
+                }
+                _ => {}
+            },
+
+            AstKind::ExportDefaultDeclaration(exp) => match &exp.declaration {
+                ExportDefaultDeclarationKind::TSInterfaceDeclaration(decl) => ctx.diagnostic(
+                    ConsistentTypeDefinitionsDiagnostic("type", "interface", decl.span),
+                ),
+                _ => {}
+            },
+
+            AstKind::TSInterfaceDeclaration(typ)
+                if self.config == ConsistentTypeDefinitionsConfig::Type =>
+            {
+                ctx.diagnostic(ConsistentTypeDefinitionsDiagnostic("type", "interface", typ.span))
+            }
+            _ => {}
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("var foo = {};", Some(serde_json::json!(["interface"]))),
+        ("interface A {}", Some(serde_json::json!(["interface"]))),
+        (
+            "
+            interface A extends B {
+              x: number;
+            }
+                  ",
+            Some(serde_json::json!(["interface"])),
+        ),
+        ("type U = string;", Some(serde_json::json!(["interface"]))),
+        ("type V = { x: number } | { y: string };", Some(serde_json::json!(["interface"]))),
+        (
+            "
+            type Record<T, U> = {
+              [K in T]: U;
+            };
+                  ",
+            Some(serde_json::json!(["interface"])),
+        ),
+        ("type T = { x: number };", Some(serde_json::json!(["type"]))),
+        ("type A = { x: number } & B & C;", Some(serde_json::json!(["type"]))),
+        ("type A = { x: number } & B<T1> & C<T2>;", Some(serde_json::json!(["type"]))),
+        (
+            "
+            export type W<T> = {
+              x: T;
+            };
+                  ",
+            Some(serde_json::json!(["type"])),
+        ),
+    ];
+
+    let fail = vec![
```

Comment:
Our rulegen didn't generate case surrounded by [`noFormat`](https://github.com/typescript-eslint/typescript-eslint/blob/fde44275117b3a7bbdcaaf267c73986b6637c5f9/packages/eslint-plugin/tests/rules/consistent-type-definitions.test.ts#L66), it will be [fixed](https://github.com/oxc-project/oxc/pull/2903)! Could you update the test cases?

Reply:
done, let me know if it looks okay

Reply:
Checked, it has been added.